addons/mrp/mrp.py

```diff
@@ -1307,8 +1308,26 @@ class mrp_production_product_line(osv.osv):
         'name': fields.char('Name', required=True),
         'product_id': fields.many2one('product.product', 'Product', required=True),
         'product_qty': fields.float('Product Quantity', digits_compute=dp.get_precision('Product Unit of Measure'), required=True),
+        'lot_id': fields.many2one('stock.production.lot', 'Lot/Serial Number'),
         'product_uom': fields.many2one('product.uom', 'Product Unit of Measure', required=True),
         'product_uos_qty': fields.float('Product UOS Quantity'),
         'product_uos': fields.many2one('product.uom', 'Product UOS'),
         'production_id': fields.many2one('mrp.production', 'Production Order', select=True),
+        'bom_line_id': fields.many2one('mrp.bom.line', 'BOM', select=True),
+    }
+    _defaults = {
+        'product_qty': 1.00
     }
+
+    def onchange_product(self, cr, uid, ids, product_id=False, context=None):
+        """ On change of product id, if finds UoM.
+        @param product_id: Changed Product id
+        """
+        if not product_id:
+            return {}
+        product = self.pool.get('product.product').browse(cr, uid, [product_id])[0]
```

Comment:
It's good practice to pass context in browse() if it's available and you can pass 'product_id' here instead of passing it in list to avoid accessing the result from 0th index.